### PR TITLE
Reorganize the strong header type extensions. Remove SendAsync extensions

### PR DESCRIPTION
**Pull request by Tratcher**
#274 
HeaderDictionaryTypeExtensions: "Remove everything except GetTypedHeaders. Everything else goes to  public method on Either ReuqestHeaders or ResponseHeaders depending on who uses them"
HttpResponseSendingExtensions: "Cut all of these. If there is significant or painful to work around uses then come back to us."

### CODE REVIEW

**Comment by Tratcher** [comment]
@muratg @davidfowl

File: src/Microsoft.AspNet.Http.Extensions/HeaderDictionaryTypeExtensions.cs [hunk 1]
```diff
@@ -23,12 +23,14 @@ public static ResponseHeaders GetTypedHeaders(this HttpResponse response)
             return new ResponseHeaders(response.Headers);
         }
 
-        public static DateTimeOffset? GetDate([NotNull] this IHeaderDictionary headers, [NotNull] string name)
+        // These are all shared helpers used by both RequestHeaders and ResponseHeaders
+
```
Comment (muratg) [inline src/Microsoft.AspNet.Http.Extensions/HeaderDictionaryTypeExtensions.cs]: nit: blank line

**Comment by muratg** [comment]
RequestHeaders and ResponseHeaders seem to share some code. Would it make sense to create an abstract class that covers the shared functionality, and move the functionality in the HeaderDictionaryTypeExtensions (which are currently becoming Internal with this change) to this shared abstract class?

**Comment by Tratcher** [comment]
I tried that in the initial feature work but it was rejected in code review.

**Comment by muratg** [comment]
:shipit:

**Pull request merged**